include/types.h

```diff
@@ -74,3 +74,65 @@ typedef struct _NetplanStateIterator NetplanStateIterator;
 struct _NetplanStateIterator {
     void* placeholder;
 };
+
+/*
+ * Errors and error domains
+ *
+ * NOTE: if new errors or domains are added,
+ * netplan/libnetplan.py must be updated with the new entries.
+ */
+
+enum NETPLAN_ERROR_DOMAINS {
+    NETPLAN_PARSER_ERROR = 1,
+    NETPLAN_VALIDATION_ERROR,
+    NETPLAN_FILE_ERROR,
+    NETPLAN_BACKEND_ERROR,
+    NETPLAN_EMITTER_ERROR,
+    NETPLAN_FORMAT_ERROR,
+};
+
+/*
+ * Errors for domain NETPLAN_PARSER_ERROR
+ *
+ * PARSER_ERRORS are expected to contain the file name, line and column numbers
+ */
+enum NETPLAN_PARSER_ERRORS {
+    NETPLAN_ERROR_INVALID_YAML,
+    NETPLAN_ERROR_INVALID_CONFIG
+};
+
+/*
+ * Errors for domain NETPLAN_VALIDATION_ERROR
+ *
+ * VALIDATION_ERRORS are expected to contain only the YAML file name
+ * where the error was found.
```

Comment:
I don't understand this error path. In what scenarios do we expect to only have a filename available during error handling and not specific guidance about the lines, columns or config xpath object that is representing a problem?